src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs

```diff
@@ -716,5 +743,56 @@ private Transaction MakeTransaction(JObject opts)
                 _ => blockTag.HexToUlong()
             };
         }
+
+        private ulong? GetGasUsed(string blockHash)
+        {
+            // Reference from eth_getBlockByHash
+            var block = _blockManager.GetByHash(blockHash.HexToBytes().ToUInt256());
+            if (block == null)
+                return null;
+
+            List<TransactionReceipt> txs = new List<TransactionReceipt>();
+            try
+            {
+                foreach (var txHash in block.TransactionHashes)
+                {
+                    Logger.LogInformation($"Transaction hash {txHash.ToHex()} in block {block.Header.Index}");
+                    TransactionReceipt? tx = _transactionManager.GetByHash(txHash);
+                    if (tx is null)
+                    {
+                        Logger.LogWarning($"Transaction not found in DB {txHash.ToHex()}");
+                    }
+                    else txs.Add(tx);
+                }
+            }
+            catch (Exception e)
+            {
+                Logger.LogWarning($"Exception {e}");
+                Logger.LogWarning($"block {block!.Hash},  {block.Header.Index}, {block.TransactionHashes.Count}");
+                foreach (var txhash in block.TransactionHashes)
+                    Logger.LogWarning($"txhash {txhash.ToHex()}");
+            }
+
+            ulong gasUsed = 0;
+            try
+            {
+                gasUsed = txs.Aggregate(gasUsed, (current, tx) => current + tx.GasUsed);
```

Comment:
We don't need this second loop if we calculate common used gas in the first loop

Reply:
sure